packages/neuron-ui/src/containers/Providers/index.tsx

```diff
@@ -20,13 +20,14 @@ const withProviders = (Comp: React.ComponentType) => (props: React.Props<any>) =
   useEffect(() => {
     ipc.asw()
   }, [])
-
   ipcRenderer.on('ASW', (_e: any, args: Response<any>) => {
-    setWallet({
-      ...wallet,
-      name: 'asw',
-      wallet: args.result,
-    })
+    if (wallet) {
```

Comment:
IMO, wallet should not be `null`, we can set it with initial values, by that much validation could be omitted.

Reply:
but I am not sure, what is the inital values when user first open app if it is not null

Reply:
Good question. Checking null is annoying. Meanwhile ‘wallet.wallet’ is strange. 

Something like ‘wallet.isPersisted’ might be good? Only when this returns true we know it’s a wallet that’s already on disk.

Reply:
Maybe `!wallet.address` or `!wallet.name` is enough, `wallet.wallet` will be refactored once the structure of the wallet is determined.

Reply:
@Keith-CY While `!wallet.address` or `!wallet.name` can deduce to the conclusion that this might not be an empty wallet, it's not proper from the usage (what if we allow a wallet to have no name in the future?). So it's simpler to just have a more meaningful method/property, even if that's only a wrap around `!wallet.address`.

👍 on the pending `wallet.wallet` refactoring.

Reply:
temporarily agree with @Keith-CY to

1.  move wallet.wallet.address to wallet.address (remove Checking null); 

1. MainContent style as "width: 100%; height: 100%;" for wizard layout; 

demo the process of creation wallet when run the app for first time( set the wallet.address as empty)